opus/application/static_media/js/cart.js

```diff
@@ -388,6 +388,19 @@ var o_cart = {
 
         o_cart.lastCartRequestNo++;
         $.getJSON(url  + add_to_url + "&reqno=" + o_cart.lastCartRequestNo, function(statusData) {
+            if (statusData.reqno < o_cart.lastCartRequestNo) {
```

Comment:
Ooooo...this is going to be a problem. While we only want to display the cart badge number for the final API call, any one of the intermediate calls might return an error that the user cares about. So you have to pop up the dialog for *all* returns, but only update the badge for the "last" one. Just move this if below the statusData if below and it should be fine.